synadm/api.py

```diff
@@ -297,7 +311,8 @@ def room_get_id(self, room_alias):
                 response as it might contain Synapse's error message.
         """
         room_directory = self.query(
-            "get", f"client/r0/directory/room/{urllib.parse.quote(room_alias)}"
+            "get", "client/r0/directory/room/{room_alias}",
+            room_alias=room_alias
```

Comment:
I think it would be good if we state this new concept somewhere in the contributing docs. Telling the contributors about the new handling of args. I'm not quite sure where this should be.

Around here a new chapter around string formatting maybe? https://github.com/JOJ0/synadm/blob/master/CONTRIBUTING.md#command-design

Reply:
not ready to merge (conflicts and outdated dev branch mess), but i'll hand off documentation to you then.